dojo_plugin/api/__init__.py

```diff
@@ -14,9 +15,34 @@
 from .v1.workspace import workspace_namespace
 from .v1.search import search_namespace
 from .v1.test_error import test_error_namespace
+from .v1.user import user_namespace
 
 api = Blueprint("pwncollege_api", __name__)
 
+# Get CORS origin from environment variable
+cors_origin = os.environ.get('CORS_ORIGINS', 'http://localhost:5173')
+
+# Add CORS headers to all responses
+@api.after_request
+def after_request(response):
+    response.headers['Access-Control-Allow-Origin'] = cors_origin
+    response.headers['Access-Control-Allow-Headers'] = 'Content-Type, Authorization'
+    response.headers['Access-Control-Allow-Methods'] = 'GET, POST, PUT, PATCH, DELETE, OPTIONS'
+    response.headers['Access-Control-Allow-Credentials'] = 'true'
+    return response
+
+# Handle OPTIONS preflight requests
+@api.before_request
+def handle_preflight():
+    if request.method == "OPTIONS":
+        response = make_response()
+        response.headers['Access-Control-Allow-Origin'] = cors_origin
+        response.headers['Access-Control-Allow-Headers'] = 'Content-Type, Authorization'
+        response.headers['Access-Control-Allow-Methods'] = 'GET, POST, PUT, PATCH, DELETE, OPTIONS'
+        response.headers['Access-Control-Allow-Credentials'] = 'true'
+        response.headers['Access-Control-Max-Age'] = '3600'
+        return response
+
```

Comment:
Maybe we change the logic so there's no default (aka default of `None`), and if there is a non-None value, then we add the CORS middleware.